### PR TITLE
Add TypeScript types

### PR DESCRIPTION
**Pull request by xcatliu**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
+// Type definitions for looks-same 4.0
+// Project: https://github.com/gemini-testing/looks-same/releases
+// Definitions by: xcatliu <https://github.com/xcatliu>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node"/>
+
+type LooksSameCallback = (error: Error | null, equal: boolean) => any;
```
Comment (DudaGod) [inline index.d.ts]: why returned type is `any` and not a `void`?
Documentation - http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html (Callback Types)

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+     */
+    ignoreAntialiasing?: boolean;
+    /**
+     * Sometimes the antialiasing algorithm can work incorrectly due to some features of the browser rendering engine.
+     * Use the option antialiasingTolerance to make the algorithm less strict.
+     * With this option you can specify the minimum difference in brightness (zero by default)
+     * between the darkest/lightest pixel (which is adjacent to the antialiasing pixel) and theirs adjacent pixels.
+     *
+     * We recommend that you don't increase this value above 10. If you need to increase more than 10 then this is definitely not antialiasing.
+     */
```
Comment (DudaGod) [inline index.d.ts]: Can you explain please why we should write all this information about each option which is already present in readme?

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+/**
+ * The options passed to looksSame.createDiff function
+ */
+interface CreateDiffOptions extends CreateDiffAsBufferOptions {
+    /**
+     * The diff image path to store
+     */
+    diff: string;
```
Comment (DudaGod) [inline index.d.ts]: Why we should create one more interface if we can set the `diff` field as optional - `diff?: string`?

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+ * @param image1 The first image path
+ * @param image2 The second image path
+ * @param callback Call when finish compare
+ */
+declare function looksSame(image1: string, image2: string, callback: LooksSameCallback): void;
+
+// https://stackoverflow.com/questions/44058101/typescript-declare-third-party-modules
+declare namespace looksSame {
+    function createDiff(options: CreateDiffOptions, callback: (error: Error | null) => any): void;
+    function createDiff(options: CreateDiffAsBufferOptions, callback: (error: Error | null, buffer: Buffer) => any): void;
```
Comment (DudaGod) [inline index.d.ts]: To my mind we don't need this overload. Do you agree?

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    /**
+     * The diff image path to store
+     */
+    diff: string;
+}
+
+/**
+ * Pass to looksSame.colors function
+ */
+interface LooksSameColor {
```
Comment (DudaGod) [inline index.d.ts]: I would just call this type `Color`

File: index.d.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    function createDiff(options: CreateDiffOptions, callback: (error: Error | null) => any): void;
+    function createDiff(options: CreateDiffAsBufferOptions, callback: (error: Error | null, buffer: Buffer) => any): void;
+
+    /**
+     * Compare two colors
+     * @param color1 The first color
+     * @param color2 The second color
+     * @param options The options passed to looksSame.colors function
+     */
+    function colors(color1: LooksSameColor, color2: LooksSameColor, options: { tolerance: number }): void;
```
Comment (DudaGod) [inline index.d.ts]: Are you sure that you don't use `export` for all of these functions inside namespace?
Documentation - http://www.typescriptlang.org/docs/handbook/namespaces.html

File: package.json [hunk 7]
```diff
@@ -14,6 +15,7 @@
     "pngjs": "^3.3.3"
   },
   "devDependencies": {
+    "@types/node": "^10.12.3",
```
Comment (DudaGod) [inline package.json]: Why we should add this dependency? Moreover if you add new dependency to `package.json` you should add it to `package.lock.json` too.

File: index.d.ts (hunk 6 above)
Comment (DudaGod) [reply index.d.ts]: `options` is not required parameter. It should be optional.

File: package.json [hunk 8]
```diff
@@ -3,6 +3,7 @@
   "version": "4.0.0",
   "description": "Pure node.js library for comparing PNG-images, taking into account human color perception.",
   "main": "index.js",
+  "types": "index.d.ts",
```
[the remark this answers is not in the document]
Comment (xcatliu) [reply package.json]: Thanks for reviewing! `index.d.ts` is used by default, but it is advisable to set the `types` field.

File: index.d.ts [hunk 9]
```diff
@@ -0,0 +1,138 @@
+// Type definitions for looks-same 4.0
+// Project: https://github.com/gemini-testing/looks-same/releases
+// Definitions by: xcatliu <https://github.com/xcatliu>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node"/>
```
[the remark this answers is not in the document]
Comment (xcatliu) [reply index.d.ts]: We use `Buffer` from `@types/node`, so this is needed

File: index.d.ts (hunk 2 above)
Comment (xcatliu) [reply index.d.ts]: Because with this comments, IDE can show these docs while coding.

File: package.json (hunk 7 above)
Comment (xcatliu) [reply package.json]: Updated `package.lock.json`

File: index.d.ts (hunk 4 above)
Comment (xcatliu) [reply index.d.ts]: With this overload, typescript can provide stronger type infer like this:

image

image

File: index.d.ts (hunk 3 above)
Comment (xcatliu) [reply index.d.ts]: Because the overloaded functions need two interface types

**Comment by xcatliu** [comment]
@DudaGod Hi I've update the code, please help to review it. Thanks!

**Comment by DudaGod** [comment]
@xcatliu thank you. We will release a new version today.

**Pull request merged**